cloudflared/config.yaml

```diff
@@ -21,10 +21,20 @@ map:
 options:
   external_hostname: null
   tunnel_name: "homeassistant"
+  strict_ssl: false
```

Comment:
I would suggest to add this option for homeassistant (`external_hostname`) and for `additional_hosts[]` to separate the different services. This would require to restructure the possible options and schema. e.g.:

```yaml
...
options:
  homeassistant:
    external_hostname: ha.example.com
    strict_ssl: false
  additional_hosts:
    - hostname: example.com
       service: example
       disableChunkedEncoding: false
       strict_ssl: false
...
```

What do you think?

Reply:
agree for the additional_hosts part. For the homeassisatant part the question remains if we need it or not which depends on whether we want to link that to the creation of the certificate as suggested by me or not

Reply:
In relation to the discussion in the other comment, the scheme would then look like this in my opinion
```yaml
...
options:
  external_hostname: ha.example.com
  additional_hosts:
    - hostname: example.com
      service: example
      disableChunkedEncoding: false
      noTLSVerify: false (set default per program logic) (schema: bool?)
      originServerName: example.com (schema: str?) 
...
```

Reply:
Two things regarding this:
1. disableChunkedEncoding would also be optional (as it is at the moment)
2. I do not think that we need originServerName. This should always be the same as the hostname, right? Unless we also want to enable the user to work with even other certificates then the Cloudflare ones....maybe leave that for the time being and add it at a second stage?

So to summarise, the definition in the config would be like this:


```
options:
  external_hostname: null
  tunnel_name: "homeassistant"
  additional_hosts: []
schema:
  external_hostname: str
  tunnel_name: str
  additional_hosts:
    - hostname: str
      service: str
      disableChunkedEncoding: bool?
      noTLSVerify: bool?
  catch_all_service: str?
  nginx_proxy_manager: bool?
  quick_tunnel: bool?
  log_level: list(trace|debug|info|notice|warning|error|fatal)?
  reset_cloudflared_files: bool?
  data_folder: list(config|share|ssl)?
  custom_config: bool?
  generate_cert: bool?
```

Reply:
I agree, good point.

Reply:
How about we leave `additional_hosts` for a second PR?